File: python/insertion.py
import util

# sort implements insertion sort algorithm. The function takes an array.
def sort(a):
	for i in range(2, len(a)):
		for k in range(i, 0, -1):
			if a[k] >= a[k-1]:
				break
			util.swap(a, k, k-1)

File: java/Insertion.java
public class Insertion {
	public static void sort(int a[]) {
		for(int i = 2; i < a.length; i++) {
			for(int k = i; k > 0 && a[k] < a[k-1]; k--)
				Util.swap(a, k, k-1);
		}
	}
}
